diff --git a/services/nucreport/app/src/service.rs b/services/nucreport/app/src/service.rs
index e155299..d8334bf 100644
--- a/services/nucreport/app/src/service.rs
+++ b/services/nucreport/app/src/service.rs
@@ -83,7 +83,7 @@ pub async fn read_file(cli: &Client, user: &str, path: &str, token: &str) -> Res
     let normalized = normalized_path.to_str().context("Invalid file path")?;

     if !token.is_empty() {
-        return read_by_file_token(cli, path, token).await;
+        return read_by_file_token(cli, normalized, token).await;
     }

     let user_dir = home_dir(user);
