diff --git a/services/nucreport/app/src/service.rs b/services/nucreport/app/src/service.rs
index 6c14374..942c650 100644
--- a/services/nucreport/app/src/service.rs
+++ b/services/nucreport/app/src/service.rs
@@ -113,7 +113,7 @@ pub async fn read_file(
     let normalized = normalized.unwrap();

     if !token.is_empty() {
-        return read_by_file_token(cli, path, token).await;
+        return read_by_file_token(cli, normalized, token).await;
     }

     let user_dir = home_dir(user);
@@ -196,12 +196,12 @@ fn index_dir(user: &String) -> Vec<String> {

 async fn read_by_file_token(
     cli: &Client,
-    path: &String,
+    path: &str,
     token: &String,
 ) -> Result<String, Box<dyn Error>> {
     cli.query_one(
         "SELECT * FROM file_paths WHERE filepath = $1 AND token = $2",
-        &[path, token],
+        &[&path, token],
     )
     .await?;

